File: services/nuldoc/nuldoc-src/components/TagList.ts
import { Config, getTagLabel } from "../config.ts";
import { Element, li, span, text, ul } from "../dom.ts";

type Props = {
  tags: string[];
  config: Config;
};

export default function TagList({ tags, config }: Props): Element {
  return ul(
    { class: "entry-tags" },
    ...tags.map((slug) =>
      li(
        { class: "tag" },
        span({ class: "tag-inner" }, text(getTagLabel(config, slug))),
      )
    ),
  );
}

File: services/nuldoc/lib/nuldoc/components/tag_list.rb
module Nuldoc
  module Components
    class TagList < DOM::HTMLBuilder
      def initialize(tags:, config:)
        super()
        @tags = tags
        @config = config
      end

      def build
        ul(class: 'entry-tags') do
          @tags.each do |slug|
            li(class: 'tag') do
              span(class: 'tag-inner') { text @config.tag_label(slug) }
            end
          end
        end
      end
    end
  end
end
